halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2ClouddriverRoService.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.distributed.kubernetes.v2;
+
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerRuntimeSettings;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.ClouddriverRoProfileFactory;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.Profile;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.ServiceSettings;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class KubernetesV2ClouddriverRoService extends KubernetesV2ClouddriverService{
+  @Autowired
+  ClouddriverRoProfileFactory clouddriverRoProfileFactory;
+
+  @Override
+  public Type getType() {
+    return Type.CLOUDDRIVER_RO;
+  }
+
+  @Override
+  public boolean isEnabled(DeploymentConfiguration deploymentConfiguration) {
+    return deploymentConfiguration.getDeploymentEnvironment().getHaServices().getClouddriver().isEnabled();
+  }
+
+  @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    String filename = "clouddriver-ro.yml";
+    String path = Paths.get(getConfigOutputPath(), filename).toString();
+    profiles.add(clouddriverRoProfileFactory.getProfile(filename, path, deploymentConfiguration, endpoints));
+    return profiles;
+  }
+
+  @Override
+  public ServiceSettings defaultServiceSettings(DeploymentConfiguration deploymentConfiguration) {
```

Comment:
Can probably cut down on duplicate code this by declaring a field `profiles` and writing one function in a superclass that appends them all to the `Settings` object.

Reply:
I couldn't find a simple way to do this because of the fact that the ```Settings``` class is different for each service. E.g., this class uses ```ClouddriverService.Settings```.

I've inlined the list creation to make it less lines.